### PR TITLE
fix buf redeclared in scrapeloop

### PR DESCRIPTION
**Pull request by arugaki**
buf Has been declared in line 785, I think it is unnecessary to declare a new variable again here.

Signed-off-by: arugakiWei <arugaki.wei@daocloud.io>

### CODE REVIEW

File: scrape/scrape.go [hunk 1]
```diff
@@ -809,7 +809,7 @@ mainLoop:
 		}
 
 		b := sl.buffers.Get(sl.lastScrapeSize).([]byte)
-		buf := bytes.NewBuffer(b)
+		buf = bytes.NewBuffer(b)
```
[the remark this answers is not in the document]
Comment (arugaki) [reply scrape/scrape.go]: oh, can I ask why don't use original buf?

[the remark this answers is not in the document]
Comment (arugaki) [reply scrape/scrape.go]: I found that it was like this in release in 2.2

[the remark this answers is not in the document]
Comment (juliusv) [reply scrape/scrape.go]: This seems to be the commit where it was forgotten to remove the original (non-pooled) buf: https://github.com/prometheus/prometheus/commit/5bed8af4cb6e74d1b9885d8de26959e9e25de861#diff-0e6adb82343526da97d01864309d918e

**Comment by juliusv** [comment]
👍 thanks!

**Pull request merged**